src/api/uploads/PlainUpload.js

```diff
@@ -79,7 +79,7 @@ class PlainUpload extends BaseUpload {
         }
 
         const attributes = JSON.stringify({
-            name: this.file.name,
+            name: this.fileName || this.file.name,
```

Comment:
I see this is used in a couple places such as in ```makePreflightRequest```. Is it possible to refactor the class a bit to always have this.fileName be the correct name?

Reply:
Here it depends on whether its subclass set this.fileName or not, and also at the beginning this.fileName is not set. I don't want to affect the existing behavior in case I break something.